cognition_objects/project.py

```diff
@@ -331,6 +332,23 @@ def update(
 
         project.macro_config = new_values
         flag_modified(project, "macro_config")
+    if llm_config is not None:
+        new_values = project.llm_config
+        if new_values is None:
+            new_values = {}
+        for key in llm_config:
+            if isinstance(llm_config[key], dict):
+                if key not in new_values:
+                    new_values[key] = {}
+                for sub_key in llm_config[key]:
+                    new_values[key][sub_key] = llm_config[key][sub_key]
```

Comment:
this now allows a max depth of two. I think for now this is fine. However with this approach we need to remember this / have to find it if problematic. 

I usually prefer options we don't need to remember since a lot of things are forgotten over time. Something like the `transferNestedDict` from the frontend could do the trick on the backend as well so we have a util method we can always use in these instances. (Not necessary for this feature but the next time a nested dict needs to be transferred i think we should invest the time)

Reply:
Comment added for extension if needed in the future